src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,34 @@
 package core.basesyntax;
 
 public class MainApp {
+    public MainApp(String[] machine) {
 
+        Machine truck = new Truck() {
+            @Override
+
+            public void doWork() {
+                super.doWork();
+            }
+        };
+        truck.doWork();
+        truck.stopWork();
+        Machine excavator = new Excavator() {
+            @Override
+
+            public void doWork() {
+                super.doWork();
+            }
+        };
```

Comment:
For `Excavator`, an abstract reference should be used as well: `Machine excavator = new Excavator();`.